REF: collect attribute-setting at the end of create_axes

jbrockmendel opened the pull request:
Ideally we wouldn't be altering `self` inplace at all, but we can make this a lot easier to reason about by collecting all of the attribute-setting at the end of the create_axes call.

`self.non_index_axes` I couldn't get all the way to the end of the call without a more invasive edit, which ill save for a separate pass to keep the diff smallish

jbrockmendel committed:
REF: collect attribute-setting at the end of create_axes

jbrockmendel committed:
Merge branch 'master' of https://github.com/pandas-dev/pandas into cln-pytables-create_axes

WillAyd commented on pandas/io/pytables.py:
```diff
@@ -3785,15 +3785,13 @@ def create_axes(
         # create according to the new data
-        self.non_index_axes = []
-        self.data_columns = []
+        nia: List = []
```
Any chance of adding a subtype here?

simonjayhawkins replied:
why the rename? I would imagine `non_index_axes` newcomer friendly than a TLA.

jbrockmendel replied:
> why the rename? I would imagine non_index_axes newcomer friendly than a TLA.

i wanted something not-identical to the self.foo name for grepping purposes, but can try to find a middle ground that is more descriptive

> Any chance of adding a subtype here?

I'll give it the ol' college try.

WillAyd commented on pandas/io/pytables.py:
```diff
@@ -3785,15 +3785,13 @@ def create_axes(
         # create according to the new data
-        self.non_index_axes = []
-        self.data_columns = []
+        nia: List = []
```
Any chance of adding a subtype here?

WillAyd reviewed (commented):
Sure on board with this

simonjayhawkins commented on pandas/io/pytables.py:
```diff
@@ -3785,15 +3785,13 @@ def create_axes(
-        self.non_index_axes = []
-        self.data_columns = []
+        nia: List = []
+        dcs = []
```
same

jbrockmendel committed:
Merge branch 'master' of https://github.com/pandas-dev/pandas into cln-pytables-create_axes

jbrockmendel committed:
more descriptive names

jbrockmendel commented:
changed to more descriptive names, added annotation for new_data_columns, non_index_axes i kept as `List`because I think we can do a better job of being consistent about what we're passing, want to wait to annotate until then

jreback commented:
sure
